lib/amqp/basic.ex

```diff
@@ -346,9 +346,17 @@ defmodule AMQP.Basic do
 
     consumer_pid = consumer_pid || self()
 
-    receiver = ReceiverManager.register_handler(chan.pid, consumer_pid, :consume)
+    pid =
+      case chan.consumer_type do
+        :selective ->
+          %{pid: pid} = ReceiverManager.register_handler(chan.pid, consumer_pid, :consume)
+          pid
 
-    case :amqp_channel.subscribe(chan.pid, basic_consume, receiver.pid) do
+        :direct ->
+          consumer_pid
```

Comment:
Do we still need to subscribe direct consumer pid? Isn't it subscribed automatically when channel is opened?

Reply:
AFAIU, `amqp_channel.subscribe/3` binds channel to specific queue (with opts specified in `#basic_consume{}`. The pid is in fact ignored by our (and rabbitmq) implementation of direct consumer, but the server method needs to be sent anyway. 
Basically we can pass any pid there, but it doesn't make sense to start extra receiver as it will not get any messages.

Reply:
Why do you want to subscribe a channel when you have a direct consumer for it?